daemon/cmd/daemon.go

```diff
@@ -357,9 +358,41 @@ func removeOldRouterState(ipv6 bool, restoredIP net.IP) error {
 		if err := netlink.AddrDel(l, &a); err != nil {
 			errs = append(errs, fmt.Errorf("failed to remove IP %s: %w", a.IP, err))
 		}
+		// Clean up old router based IP rules installed for encryption. See https://github.com/cilium/cilium/pull/14924
+		// Previously, these rules were always installed in ENI mode. This clean up is not gated with IPsec to handle
+		// for cases where a rollout disables IPsec or if user upgrades from prior versions.
+		if option.Config.IPAM == ipamOption.IPAMENI {
+			err := removeOldRouterIPRules(a.IP)
+			if err != nil {
+				errs = append(errs, fmt.Errorf("failed to remove IP rule for old router IP %s: %w", a.IP, err))
+			}
+		}
+	}
+	if len(errs) > 0 {
+		return fmt.Errorf("failed to clean up old router state: %v", errs)
+	}
+
+	return nil
+}
+
+// removeOldRouterIPRules cleans up IP rules installed for the old router IP.
+func removeOldRouterIPRules(ip net.IP) error {
+	info := node.GetRouterInfo()
```

Comment:
Can you validate that this call here is not called too early where the node doesn't yet have the router info? And then could you leave a comment as to why it's OK?

Reply:
Aah ! I thought I validated that router info was available at this stage. Looks like it isn't. Because it's being set only in `d.allocateIPs()` which is called after `d.restoreCiliumHostIPs()`

I can actually reuse information from `d.ipam.GetVpcCIDRs()` which is already being called in `restoreCiliumHostIPs()`

@christarazi does that sound okay ?

Reply:
Yep, as long as the dependencies line up, there should be no problem.